version: "3.8"
services:
    db:
        image: postgres:15.1-alpine
        container_name: db_app
        volumes:
            - pgdata:/var/lib/postgresql/data
        expose:
            - 1221
        ports:
            - "1221:5432"
        env_file:
            - .env-prod
        networks:
            - django_app-network
        healthcheck:
            test: ["CMD-SHELL", "pg_isready -U postgres"]
            interval: 5s
            timeout: 5s
            retries: 5
    app:
        build:
            context: .
            dockerfile: Dockerfile
        env_file:
            - .env-prod
        container_name: django_app
        command: ["/django_app/docker/app.sh"]
        volumes:
            - /app/generated_files
        networks:
            - django_app-network
        ports:
            - 8000:8000
        depends_on:
            db:
                condition: service_healthy
        restart: always

networks:
    django_app-network:
        name: django_app-network
volumes:
    pgdata: